ui/desktop/src/components/ReportFailureModal.tsx

```diff
@@ -0,0 +1,285 @@
+import { useState, useEffect, useCallback } from 'react';
+import { Button } from './ui/button';
+import { Textarea } from './ui/textarea';
+import { Card, CardContent } from './ui/card';
+import { AlertCircle, CheckCircle, Loader2 } from 'lucide-react';
+import Modal from './Modal';
+import { getSecretKey } from '../config';
+import { getExtensions, providers } from '../api/sdk.gen';
+
+interface SystemInfo {
+  gooseVersion: string;
+  osVersion: string;
+  platform: string;
+  architecture: string;
+  providerType?: string;
+  extensionCount: number;
+}
+
+interface ReportFailureModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function ReportFailureModal({ isOpen, onClose }: ReportFailureModalProps) {
+  const [description, setDescription] = useState('');
+  const [systemInfo, setSystemInfo] = useState<SystemInfo | null>(null);
+  const [recentErrors, setRecentErrors] = useState<string[]>([]);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [submitStatus, setSubmitStatus] = useState<'idle' | 'success' | 'error'>('idle');
+
+  const getExtensionCount = async (): Promise<number> => {
+    try {
+      const secretKey = getSecretKey();
+      const response = await getExtensions({
+        headers: secretKey ? { 'X-Secret-Key': secretKey } : {},
+      });
+
+      if (response.data && Array.isArray(response.data)) {
+        return response.data.length;
+      }
+    } catch (error) {
+      console.error('Failed to get extension count:', error);
+    }
+    return 0;
+  };
+
+  const collectSystemInfo = useCallback(async () => {
+    try {
+      const secretKey = getSecretKey();
+      const info: SystemInfo = {
+        gooseVersion: String(
+          window.appConfig?.get('GOOSE_VERSION') ||
+            window.appConfig?.get('version') ||
+            'Development'
+        ),
+        osVersion: navigator.userAgent,
+        platform: window.electron?.platform || 'Unknown',
+        architecture: window.electron?.arch || 'Unknown',
+        extensionCount: await getExtensionCount(),
+      };
+
+      // Try to get current provider type (without sensitive data)
+      try {
+        const providerResponse = await providers({
+          headers: secretKey ? { 'X-Secret-Key': secretKey } : {},
+        });
+
+        if (
+          providerResponse.data &&
+          Array.isArray(providerResponse.data) &&
+          providerResponse.data.length > 0
+        ) {
+          info.providerType = providerResponse.data[0].name || 'Unknown Provider';
```

Comment:
I don't think you want just the first provider.

The right thing to use here is probably the session metadata for the active session. #4648 adds that to the client API, so this might fit in better once that's merged